### ISSUE TITLE
Neovim 0.9 does the opposite for Editorconfig's insert_final_newline 

### ISSUE THREAD

**Issue by viktorsmari** [neovim/neovim#21648]
### Neovim version (nvim -v)

NVIM v0.9.0-dev-613+gef18c9f9b

### Vim (not Nvim) behaves the same?
No it does the opposite. It does not change the last line in the file.
VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Aug 23 2022 20:18:58)

### Operating system/version

Ubuntu 22.10

### Terminal name/version

xfce4

### $TERM environment variable

xterm-256color

### Installation

Appimage

### How to reproduce the issue

Create an `.editorconfig` file with the contents

```bash
[*]
insert_final_newline = false
```

Open a file and save it. I use `vim-gitgutter` that tells me that the last line has been changed when I open the file.

![image](https://user-images.githubusercontent.com/1689020/210649972-2540e401-80d4-49f3-ae38-e34bf26d9067.png)

![image](https://user-images.githubusercontent.com/1689020/210652985-3cf955eb-fa43-4bec-bc23-29e2870721d3.png)

### Expected behavior

Do not change the last line of the file?
It did not change the last line in neovim 0.6

### Actual behavior

It changes the last line of the file, in every file it opens.

In my last neovim version (0.6) this did not happen.
This also does not happen in vim.

If I remove the `.editorconfig` file, the files are not changed.

**Comment by fsouza** [comment]
`insert_final_newline = false` means the editor is supposed to ensure the file doesn't end with a newline (see spec at https://editorconfig.org/ - search for insert_final_newline, unfortunately I can't deep link it).

neovim is correctly removing the `\n` at the end of the file.

**Comment by viktorsmari** [comment]
Thanks @fsouza - Did neovim 0.6 then not support the editorconfig?
I was using the `Plug 'editorconfig/editorconfig-vim'` before and I **thought** it was working before :thinking:

**Comment by fsouza** [comment]
Native support for editorconfig is brand new.

Looking at editorconfig-vim (https://github.com/editorconfig/editorconfig-vim/blob/30ddc057f71287c3ac2beca876e7ae6d5abe26a0/plugin/editorconfig.vim#L461-L473), they're not doing the same thing as neovim.

And it appears that my interpretation of the spec is incorrect: https://github.com/editorconfig/editorconfig/issues/475 leads me to think that the spec is poorly worded and the behavior of editorconfig-vim is the correct one.

There's an issue in editorconfig-vim too: https://github.com/editorconfig/editorconfig-vim/issues/136

**Comment by fsouza** [comment]
@zeertzjq according to https://github.com/editorconfig/editorconfig/issues/475, this issue may actually be valid 🤔

**Comment by gpanders** [comment]
I think we stick to what the spec says. If the EditorConfig people decide the spec is wrong and change it, we can change it to (it's as simple as removing the modification to the `'endofline'` option).

**Comment by justinmk** [comment]
Does https://github.com/neovim/neovim/pull/22584 look correct?

**Comment by meldafert** [comment]
I was more arguing along the lines of adding an option like `set formatoptions-=t`, which makes neovim ignore the `max_line_length` editorconfig setting, regardless of whether the spec thinks the default should be to automatically correct it or not.

**Comment by gpanders** [comment]
```lua
require('editorconfig').insert_final_newline = nil
```

will disable this behavior completely, and is pretty easy to add to one's config (it's not _that_ much longer than setting an option).

**Comment by meldafert** [comment]
@gpanders Yes, something like this would be helpful - but unfortunately, this does not work for me with either 0.9.1 or v0.10.0-dev-535+g8c9dab3e0.

**Comment by gpanders** [comment]
> (Actually, it works if I set `require('editorconfig').properties.insert_final_newline = nil`. Thanks for the pointer!)

Thanks, my mistake. I updated my original comment.
